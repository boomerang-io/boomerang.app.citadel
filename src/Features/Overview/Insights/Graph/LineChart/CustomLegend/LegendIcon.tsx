import React from "react";

type Props = {
    className?: string;
    strokeColor?: string;
};

const LegendIcon = ({ className, strokeColor }: Props) => {
  return (
    // @ts-expect-error ts-migrate(17004) FIXME: Cannot use JSX unless the '--jsx' flag is provided... Remove this comment to see the full error message
    <svg
      className={className}
      width="100%"
      height="100%"
      viewBox="10 0 13 28"
      version="1.1"
      xmlns="http://www.w3.org/2000/svg"
    >
      {/* @ts-expect-error ts-migrate(17004) FIXME: Cannot use JSX unless the '--jsx' flag is provided... Remove this comment to see the full error message */}
      <path
        strokeWidth="4"
        fill="none"
        stroke={strokeColor}
        d="M0,16h10.666666666666666
          A5.333333333333333,5.333333333333333,0,1,1,21.333333333333332,16
          H32M21.333333333333332,16
          A5.333333333333333,5.333333333333333,0,1,1,10.666666666666666,16"
        className="recharts-legend-icon"
      />
    </svg>
  );
};

export default LegendIcon;
