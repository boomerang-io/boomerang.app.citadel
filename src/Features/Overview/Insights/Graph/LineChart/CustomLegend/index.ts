// @ts-expect-error ts-migrate(6142) FIXME: Module './CustomLegend' was resolved to 'C:/Users/... Remove this comment to see the full error message
export { default } from "./CustomLegend"