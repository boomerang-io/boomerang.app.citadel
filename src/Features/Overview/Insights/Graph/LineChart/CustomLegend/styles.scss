@import "Styles/variables";

.c-legend {
  display: flex;
  flex-direction: column;
  justify-content: space-between;
  height: 2rem;
  margin-right: -3rem;
}
.c-legend-icon {
  display: flex;
  align-items: center;
  cursor: pointer;
}
.b-legend-icon {
  background-color: transparent;
  width: 1.875rem;
  height: 1rem;
  color: $text-02;
}
.--toggled {
  opacity: 0.4;
}
.b-legend-label {
  margin-right: 1rem;
}
