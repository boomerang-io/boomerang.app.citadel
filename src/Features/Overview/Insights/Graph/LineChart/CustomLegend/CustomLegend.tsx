import React, { Component } from "react";
// @ts-expect-error ts-migrate(7016) FIXME: Could not find a declaration file for module '@boo... Remove this comment to see the full error message
import { isAccessibleEvent } from "@boomerang-io/utils";
// @ts-expect-error ts-migrate(6142) FIXME: Module './LegendIcon' was resolved to 'C:/Users/Is... Remove this comment to see the full error message
import LegendIcon from "./LegendIcon";
import "./styles.scss";

type Props = {
    payload?: any[];
    toggleItem?: (...args: any[]) => any;
    toggledItems?: any[];
    lastDots?: any[];
};

class CustomLegend extends Component<Props> {

  toggleWithIsAccessibleEventCheck = (e: any, data: any) => {
    // @ts-expect-error ts-migrate(2722) FIXME: Cannot invoke an object which is possibly 'undefin... Remove this comment to see the full error message
    isAccessibleEvent(e) && this.props.toggleItem(data);
  };
  render() {
    return (
      // @ts-expect-error ts-migrate(17004) FIXME: Cannot use JSX unless the '--jsx' flag is provided... Remove this comment to see the full error message
      <div className="c-legend">
        {/* @ts-expect-error ts-migrate(2532) FIXME: Object is possibly 'undefined'. */}
        {this.props.payload.map((data, index) => {
          // @ts-expect-error ts-migrate(2532) FIXME: Object is possibly 'undefined'.
          let isToggled = this.props.toggledItems.find((item) => data.payload.name === item);
          // @ts-expect-error ts-migrate(2532) FIXME: Object is possibly 'undefined'.
          let lastDot = this.props.lastDots.find((dot) => dot.key === data.payload.dataKey);
          return (
            // @ts-expect-error ts-migrate(17004) FIXME: Cannot use JSX unless the '--jsx' flag is provided... Remove this comment to see the full error message
            <div
              key={`${data.payload.name}-${index}`}
              className="c-legend-icon"
              onClick={(e) => this.toggleWithIsAccessibleEventCheck(e, data)}
              onKeyDown={(e) => this.toggleWithIsAccessibleEventCheck(e, data)}
              role="button"
              // @ts-expect-error ts-migrate(2322) FIXME: Type 'string' is not assignable to type 'number | ... Remove this comment to see the full error message
              tabIndex="0"
              style={{
                position: "relative",
                left: "2rem",
                top: lastDot ? lastDot.y : "",
                marginTop: "-1.75rem",
                padding: "0.25rem",
              }}
            >
              {/* @ts-expect-error ts-migrate(17004) FIXME: Cannot use JSX unless the '--jsx' flag is provided... Remove this comment to see the full error message */}
              <LegendIcon className="b-legend-icon" strokeColor={data.payload.stroke} />
              {/* @ts-expect-error ts-migrate(17004) FIXME: Cannot use JSX unless the '--jsx' flag is provided... Remove this comment to see the full error message */}
              <span className={`b-legend-label${isToggled ? " --toggled" : ""}`} key={data.dataKey}>
                {data.payload.name}
              </span>
            </div>
          );
        })}
      </div>
    );
  }
}

export default CustomLegend;
