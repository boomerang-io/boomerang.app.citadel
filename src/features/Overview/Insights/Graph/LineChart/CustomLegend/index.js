import React, { Component } from "react";
import PropTypes from "prop-types";
import { isAccessibleEvent } from "utils";
import LegendIcon from "./LegendIcon";
import "./styles.scss";

class CustomLegend extends Component {
  toggleWithIsAccessibleEventCheck = (e, data) => {
    isAccessibleEvent(e) && this.props.toggleItem(data);
  };
  render() {
    return (
      <div className="c-legend">
        {this.props.payload.map((data, index) => {
          let isToggled = this.props.toggledItems.find(item => data.payload.name === item);
          let lastDot = this.props.lastDots.find(dot => dot.key === data.payload.dataKey);
          return (
            <div
              key={`${data.payload.name}-${index}`}
              className="c-legend-icon"
              onClick={e => this.toggleWithIsAccessibleEventCheck(e, data)}
              onKeyDown={e => this.toggleWithIsAccessibleEventCheck(e, data)}
              role="button"
              tabIndex="0"
              style={{
                position: "relative",
                left: "2rem",
                top: lastDot ? lastDot.y : "",
                marginTop: "-1.75rem",
                padding: "0.25rem"
              }}
            >
              <LegendIcon className="b-legend-icon" strokeColor={data.payload.stroke} />
              <span className={`b-legend-label${isToggled ? " --toggled" : ""}`} key={data.dataKey}>
                {data.payload.name}
              </span>
            </div>
          );
        })}
      </div>
    );
  }
}

CustomLegend.propTypes = {
  payload: PropTypes.array,
  toggleItem: PropTypes.func,
  toggledItems: PropTypes.array,
  lastDots: PropTypes.array
};

export default CustomLegend;
