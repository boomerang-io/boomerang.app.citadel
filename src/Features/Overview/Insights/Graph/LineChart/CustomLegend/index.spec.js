import React from "react";
import CustomLegend from "./index";

const payload = [
  {
    color: "#047cc0",
    dataKey: "total",
    fill: "url(#blueGradient)",
    fillOpacity: 0.6,
    formatter: undefined,
    name: "total",
    payload: { date: 1542160248412, failed: 5, success: 12, total: 17 },
    stroke: "#047cc0",
    unit: undefined,
    value: 17
  }
];
const mockfn = jest.fn();
const toggleItem = mockfn;
const toggledItems = [];
const lastDots = [];

describe("CustomLegend --- Snapshot", () => {
  it("Capturing Snapshot of CustomLegend", async () => {
    const { baseElement } = global.rtlRouterRender(<CustomLegend payload={payload} toggleItem={toggleItem} toggledItems={toggledItems} lastDots={lastDots} />);
    expect(baseElement).toMatchSnapshot();
  });
});
