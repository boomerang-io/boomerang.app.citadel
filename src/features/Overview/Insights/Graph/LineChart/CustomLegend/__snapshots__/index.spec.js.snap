// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`CustomLegend --- Snapshot Capturing Snapshot of CustomLegend 1`] = `
<div
  className="c-legend"
>
  <div
    className="c-legend-icon"
    onClick={[Function]}
    onKeyDown={[Function]}
    role="button"
    style={
      Object {
        "left": "2rem",
        "marginTop": "-1.75rem",
        "padding": "0.25rem",
        "position": "relative",
        "top": "",
      }
    }
    tabIndex="0"
  >
    <svg
      className="b-legend-icon"
      height="100%"
      version="1.1"
      viewBox="10 0 13 28"
      width="100%"
      xmlns="http://www.w3.org/2000/svg"
    >
      <path
        class="recharts-legend-icon"
        d="M0,16h10.666666666666666 A5.333333333333333,5.333333333333333,0,1,1,21.333333333333332,16 H32M21.333333333333332,16 A5.333333333333333,5.333333333333333,0,1,1,10.666666666666666,16"
        fill="none"
        stroke-width="4"
      />
    </svg>
    <span
      className="b-legend-label"
    />
  </div>
</div>
`;
